packages/near-fast-auth-signer/src/components/SocialLogin/SocialButton.tsx

```diff
@@ -0,0 +1,147 @@
+import { FacebookAuthProvider, GoogleAuthProvider, signInWithPopup } from 'firebase/auth';
+import React from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { setAccountIdToController } from '../../lib/controller';
+import FirestoreController from '../../lib/firestoreController';
+import { extractQueryParams } from '../../utils';
+import { networkId } from '../../utils/config';
+import { deleteUserAccount, firebaseAuth } from '../../utils/firebase';
+import { storePassKeyAsFAK } from '../../utils/passkey';
+import { getSocialLoginAccountId } from '../../utils/string';
+import { onCreateAccount, onSignIn } from '../AuthCallback/auth';
+
+const Button = styled.button`
+  font-size: 14px;
+  font-weight: 600;
+  color: var(--sand12);
+  background: var(--sand3);
+  padding: 8px;
+  border: 1px solid var(--sand6);
+  display: flex;
+  align-items: center;
+  width: calc(50% - 4px) !important;
+  justify-content: center;
+  border-radius: 100px;
+  box-sizing: border-box;
+
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+
+  ${(props) => (props.disabled
+    && `
+    cursor: not-allowed;
+    opacity: 0.5;
+    color: #72727A;
+    `
+  )};
+
+`;
+
+type SocialButtonProps = {
+  provider: GoogleAuthProvider | FacebookAuthProvider;
+  logoComponent: React.ReactNode;
+  label: string;
+  isRecovery: boolean;
+  disabled?: boolean;
+};
+
+const shouldCreateAccount = (createdAt: string, lastLoginAt: string): boolean => {
+  const createdTime = new Date(createdAt).getTime();
+  const lastLoginTime = new Date(lastLoginAt).getTime();
+
+  const diffInMilliseconds = lastLoginTime - createdTime;
+  if (diffInMilliseconds === 0) return true;
+
+  const diffInMinutes = diffInMilliseconds / (1000 * 60);
+
+  return diffInMinutes < 1;
+};
+
+function SocialButton({
+  provider,
+  logoComponent,
+  label,
+  isRecovery,
+  disabled,
+}: SocialButtonProps) {
```

Comment:
Either way is fine, just sharing knowledge, but you can type a component with:

`const SocialButton: React.FC<SocialButtonProps> = (props) => {....}`

So we can improve a little bit the type of the function.